packages/salesforcedx-vscode-core/src/commands/util/postconditionCheckers.ts

```diff
@@ -32,6 +32,35 @@ import { PathStrategyFactory } from './sourcePathStrategies';
 type OneOrMany = LocalComponent | LocalComponent[];
 type ContinueOrCancel = ContinueResponse<OneOrMany> | CancelResponse;
 
+export class CompositePostconditionChecker<T> implements PostconditionChecker<T> {
+  private readonly postcheckers: Array<PostconditionChecker<any>>;
+  public constructor(...postcheckers: Array<PostconditionChecker<any>>) {
+    this.postcheckers = postcheckers;
+  }
+  public async check(inputs: CancelResponse | ContinueResponse<T>): Promise<CancelResponse | ContinueResponse<T>> {
+    if (inputs.type === 'CONTINUE') {
+      const aggregatedData: any = {};
+      for (const postchecker of this.postcheckers) {
+        const input = await postchecker.check(inputs);
+        if (input.type === 'CONTINUE') {
+          Object.keys(input.data).map(
```

Comment:
let's use forEach since we intend to use side effects here

Reply:
Are we able to use forEach given that there is an await being used inside the loop? If a simple switch to forEach is not an option is there anything else you would recommend?

Reply:
Realized I misunderstood the comment! Changed in the latest commit